dms/models/dms_file.py

```diff
@@ -377,10 +378,15 @@ def _compute_extension(self):
         for record in self:
             record.extension = file.guess_extension(record.name)
 
-    @api.depends("content")
+    @api.depends("content", "name")
     def _compute_mimetype(self):
+        """ Similar to addons/base/models/ir_attachment.py#L232 """
         for record in self:
-            record.res_mimetype = guess_mimetype(base64.b64decode(record.content or ""))
+            record.res_mimetype = mimetypes.guess_type(record.name or "")[0]
+            if record.res_mimetype == "application/octet-stream":
+                record.res_mimetype = guess_mimetype(
+                    base64.b64decode(record.content or "")
+                )
```

Comment:
IIUC, you're trying to get the mimetype from the file name and, later, from the file contents.

IMHO it should be the opposite. The contents are usually more reliable than the file extension to know the kind of file you have.